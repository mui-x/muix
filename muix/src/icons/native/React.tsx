import createSvgIcon from './lib/create-svg-icon'

export default createSvgIcon(
  'react',
  'React',
  true
)