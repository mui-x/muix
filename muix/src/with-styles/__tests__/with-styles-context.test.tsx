/** @jsx platform.createElement */
import React from 'react'
import { platform } from 'reactxx-sheeter'

import { initPlatform, mount, ReactAny } from "reactxx-tests";

describe("WITH STYLES CONTEXT", () => {
  it("WEB, with provider", () => doTest(true, true));
  it("WEB, without provider", () => doTest(true, false));
  it("NATIVE, with provider", () => doTest(false, true));
  it("NATIVE, without provider", () => doTest(false, false));
});

const ctx = React.createContext(123)
const doTest = (isWeb: boolean, withProvider: boolean) => {
  const inner = <React.Fragment>
    <ReactAny>Hallo</ReactAny>
    <ReactAny>
      <ctx.Consumer>
        {num => <ReactAny id={num.toString()} />}
      </ctx.Consumer>
    </ReactAny>
  </React.Fragment>
  const comp = mount(withProvider ? <ctx.Provider value={1}>
    {inner}
  </ctx.Provider> : inner)
  expect(comp).toMatchSnapshot();
}
