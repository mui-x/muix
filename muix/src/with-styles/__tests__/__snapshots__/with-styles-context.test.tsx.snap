// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WITH STYLES CONTEXT NATIVE, with provider 1`] = `
<div>
  <div>
    Hallo
  </div>
  <div>
    <div
      id="1"
    />
  </div>
</div>
`;

exports[`WITH STYLES CONTEXT NATIVE, without provider 1`] = `
<div>
  <div>
    Hallo
  </div>
  <div>
    <div
      id="123"
    />
  </div>
</div>
`;

exports[`WITH STYLES CONTEXT WEB, with provider 1`] = `
<div>
  <div>
    Hallo
  </div>
  <div>
    <div
      id="1"
    />
  </div>
</div>
`;

exports[`WITH STYLES CONTEXT WEB, without provider 1`] = `
<div>
  <div>
    Hallo
  </div>
  <div>
    <div
      id="123"
    />
  </div>
</div>
`;
